File: Java/playwright-tests/src/main/java/com/qa/portfolio/pages/CartPage.java
package com.qa.portfolio.pages;

import com.microsoft.playwright.Page;

import java.util.List;

public class CartPage extends BasePage {

    private static final String CART_ITEMS      = ".cart_item";
    private static final String ITEM_NAMES      = ".inventory_item_name";
    private static final String CHECKOUT_BTN    = "#checkout";
    private static final String CONTINUE_BTN    = "#continue-shopping";

    public CartPage(Page page) {
        super(page);
    }

    public boolean isLoaded() {
        return getCurrentUrl().contains("cart");
    }

    public int getItemCount() {
        return locate(CART_ITEMS).count();
    }

    public List<String> getItemNames() {
        return locate(ITEM_NAMES).allTextContents();
    }

    public CheckoutPage proceedToCheckout() {
        locate(CHECKOUT_BTN).click();
        return new CheckoutPage(page);
    }

    public InventoryPage continueShopping() {
        locate(CONTINUE_BTN).click();
        return new InventoryPage(page);
    }
}

File: CSharp/Selenium.WebTests/Pages/CartPage.cs
using OpenQA.Selenium;

namespace Selenium.WebTests.Pages;

public class CartPage : BasePage
{
    private static readonly By CartItems      = By.ClassName("cart_item");
    private static readonly By CheckoutButton = By.Id("checkout");
    private static readonly By ContinueButton = By.Id("continue-shopping");
    private static readonly By ItemNames      = By.ClassName("inventory_item_name");

    public CartPage(IWebDriver driver) : base(driver) { }

    public bool IsLoaded => CurrentUrl.Contains("cart");

    public int ItemCount => Driver.FindElements(CartItems).Count;

    public IReadOnlyList<string> ItemNames_ =>
        Driver.FindElements(ItemNames).Select(e => e.Text).ToList();

    public CheckoutPage ProceedToCheckout()
    {
        WaitForElement(CheckoutButton).Click();
        return new CheckoutPage(Driver);
    }

    public InventoryPage ContinueShopping()
    {
        WaitForElement(ContinueButton).Click();
        return new InventoryPage(Driver);
    }
}
